plugins/single_plugins/idle.cpp

```diff
@@ -223,16 +266,28 @@ class wayfire_idle_singleton : public wf::singleton_plugin_t<wayfire_idle>
         auto fs_views = output->workspace->get_promoted_views(
             output->workspace->get_current_workspace());
 
+        /* Currently, the fullscreen count would always be 0 or 1,
+         * since fullscreen-layer-focused is only emitted on changes between 0 and 1
+         **/
         if (fs_views.size())
         {
-            get_instance().idle_inhibit();
+            get_instance().inc_fullscreen();
         }
 
         screensaver_timeout.set_callback([=] ()
         {
             create_screensaver_timeout(screensaver_timeout);
         });
         create_screensaver_timeout(screensaver_timeout);
+
+        on_new_inhibitor.set_callback([&] (void *data)
+        {
+            auto inhibitor = static_cast<wlr_idle_inhibitor_v1*>(data);
+            /* will be freed by the destroy request */
```

Comment:
This will likely crash if an inhibitor is created and in the meantime the idle plugin is unloaded.